search/ui/class-vip-search-widget.php

```diff
@@ -0,0 +1,454 @@
+<?php
+
+namespace Automattic\VIP\Search\UI;
+
+/**
+* Class VIP_Search_Widget
+ * @package Automattic\VIP\Search\UI
+ *
+ * Derived from Jetpack_Search_Widget
+ */
+class VIP_Search_Widget extends \WP_Widget {
+
+	/**
+	 * Default sort order for search results.
+	 *
+	 * @var string
+	 */
+	const DEFAULT_SORT = 'relevance_desc';
+
+	public function __construct( $name = null ) {
+		if ( empty( $name ) ) {
+			$name = 'Search (VIP Enterprise)';
+		}
+
+		parent::__construct(
+			'vip-search-widget',
+			$name,
+			array( 'description' => 'UI for VIP Enterprise Search' )
+		);
+
+		if ( ! is_admin() ) {
+			add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_frontend_scripts' ) );
+		}
+	}
+
+	/**
+	 * Enqueue scripts and styles for the frontend.
+	 *
+	 */
+	public function enqueue_frontend_scripts() {
+		if ( ! is_active_widget( false, false, $this->id_base, true ) ) {
+			return;
+		}
+
+		wp_enqueue_script(
+			'vip-search-widget',
+			plugins_url( 'js/search-widget.js', __FILE__ ),
+			array(),
+			1.0,
+			true
+		);
+
+		wp_enqueue_style( 'vip-search-widget', plugins_url( 'css/search-widget-frontend.css', __FILE__ ), array(), 1.0 );
```

Comment:
Is there any particular reason the JS file is named as `search-widget.js` but the CSS one is `search-widget-frontend.css`?

Reply:
I'm just trying to stick with Jetpack's names. They have the following files (some of which we don't need yet): `search-widget-admin-ui.css`, `search-widget-frontend.css`, `search-widget.js` and `search-widget-admin.js`.